docs/ql-documentation/learn-ql/cpp/introduce-libraries-cpp.rst

```diff
@@ -4,114 +4,710 @@ Introducing the C/C++ libraries
+| Declaration syntax                                                                                                                                                                                                                                                                                                                                                                                                                          | QL class                                                                                                                                                              | Remarks                                                                                                                                                                                                                                                                           |
++=============================================================================================================================================================================================================================================================================================================================================================================================================================================+=======================================================================================================================================================================+===================================================================================================================================================================================================================================================================================+
+| ``enum`` ``{`` *enum_constant1* ``,`` *enum_constant2* ``,`` *enum_constant3* ``}``                                                                                                                                                                                                                                                                                                                                                         | `EnumConstant <https://help.semmle.com/qldoc/cpp/semmle/code/cpp/Enum.qll/type.Enum$EnumConstant.html>`__                                                             |                                                                                                                                                                                                                                                                                   |
```

Comment:
What's the simplest way for me to see what this will look like when it's rendered?

How complete is this PR at the moment?

Reply:
I'd say it is 80% done.  Formatting is totally inconsistent and still a bit in the air as far as what conventions should be used (feedback welcome).  I've pasted the link to the file below.